netket/graph/abstract_graph.py

```diff
@@ -0,0 +1,56 @@
+import abc
+import numpy as _np
+
+class PyAbstractGraph(abc.ABC):
+	"""Abstract class for NetKet graph objects"""
+
+	@property
+	@abc.abstractmethod
+	def n_sites(self):
+		r"""int: The number of vertices in the graph"""
+		return NotImplementedError
```

Comment:
This method is a duplicate of `size`. We should decide on one version.

Reply:
yes I would remove n_sites and keep only size

Reply:
I noticed that you use size to refer to the number of nodes, but in mathematics the size of the graph is normally the number of edges. What notation would you like to keep?

Reply:
yes that's a good point, networkx.size() actually returns the number of edges... Strictly speaking to be more consistent maybe we should do size -> order and/or size -> n_vertices , but I wouldn't leave the size method to return the number of edges, since this would cause an infinite number of bugs (at least on my side) because of this ambiguity (also it seems that there are multiple mathematical definitions for size).

Reply:
Alright, I will leave `size` for n_sites, then.